R/Connection.R

```diff
@@ -27,52 +27,30 @@
 #'
 #' @export
 getEunomiaConnectionDetails <- function() {
-  details <- getConnectionDetails(datasetName = "GiBleed")
-  return(details)
+  rlang::inform('getEunomiaConnectionDetails() will be deprecated in a future release
```

Comment:
Why deprecate this function? Seems like there's no harm in having this function for DatabaseConnector users?

Reply:
The reason to deprecate it is that `getEunomiaConnectionDetails` requires Java. If users switch to this syntax we can remove Java as a dependency of the Eunomia package. It's a significant reduction in the dependencies.

`createConnectionDetails(dbms = "sqlite", server = getDatabaseFile())`

Reply:
All the Book of OHDIS exercises use this function (e.g. [here](https://ohdsi.github.io/TheBookOfOhdsi/SqlAndR.html#exercises-4), so we would be breaking all those. (And we'd also need to fix all HADES package unit tests, which is doable, but we could spend those cycles on more important things)

And pretty much all HADES packages depend on Java, so there's no real gain in removing that dependency.

Hence, I'm strongly in favor of keeping this function.

Reply:
Ok that makes sense. I would like to make the Eunomia functionality (i.e. downloading example cdm datasets) available on CRAN without a dependency on Java. Does anyone object if I do this in a separate non-Hades Apache licensed R package that I'll maintain?

Reply:
If DatabaseConnector is in the Suggests section, isn't that enough? You would be able to install the package without Java, and would make it similar to CDMConnector.

Reply:
Yep exactly. This PR includes DatabaseConnector under Suggests because it is needed to run tests. 
Keep in mind that this PR should not break existing code. It just prints a message informing users that `getEunomiaConnectionDetails` will be deprecated at some point in the future and suggests a replacement.

Reply:
Yes, putting DatabaseConnector under Suggests would be fine for keeping backwards compatibility. I was just suggesting we don't print the deprecation message (and not deprecate this function in the near term). 

We may want to add some code that ensures DatabaseConnector is installed when someone calls this function, like we do [here](https://github.com/OHDSI/CohortMethod/blob/main/R/Viewer.R#L168).

Reply:
Ok yes this seems like a good solution. I'll make the change.

Reply:
Made the update. getEunomiaConnectionDetails is optional in this PR. I think this should work on CRAN.